packages/pigeon/lib/swift_generator.dart

```diff
@@ -668,19 +683,15 @@ String _camelCase(String text) {
 }
 
 String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
-  final String forceUnwrap = type.isNullable ? '' : '!';
-  final String castUnwrap = type.isNullable ? '?' : '';
   if (isEnum(root, type)) {
-    final String nullableConditionPrefix =
-        type.isNullable ? '$value == nil ? nil : ' : '';
-    return '$nullableConditionPrefix${_swiftTypeForDartType(type)}(rawValue: $value as! Int)$forceUnwrap';
+    return '${_swiftTypeForDartType(type)}(rawValue: $value as! Int)!';
   } else if (type.baseName == 'Object') {
     // Special-cased to avoid warnings about using 'as' with Any.
     return value;
-  } else if (type.baseName == 'int') {
-    return '($value is Int) ? Int64($value as! Int) : $value as! Int64$castUnwrap';
```

Comment:
swift can cast all ints to int64, so the previous logic didn't actually do anything. The integration tests cover this in every possible form, so I'm fairly certain this is good as is.

Reply:
We can't **cast** across different int types (since they differ not just in type, but in actual underlying bit representation). We have to **convert** it using one of its constructors. 

You can try out a quick experiment in Swift playground to get an error: 
```
let int32: Int32 = 88
let int64 = int32 as! Int64
print(int64)
```

What are the different types `args[0]` can possibly have? (there are like ~10 Int-related types in Swift).

Reply:
the thing here is that all incoming types are `Any` so they can all be converted directly to int64

Reply:
but I'm going to make it work for any int type, it's really not a big deal to add the little bit of extra code needed to make that happen

Reply:
> the thing here is that all incoming types are `Any` so they can all be converted directly to int64

I think you're confusing variable types and object instance types here. The deserialized objects have an actual type at runtime, even if you tell the compiler to pretend they don't at compile time by casting to `Any`.

Taking the example above, for instance, and changing it to:
```
let int32: Int32 = 88
let int64 = (int32 as! Any) as! Int64 // ERROR
print(int64)
```

throws `Could not cast value of type 'Swift.Int32' (<address>) to 'Swift.Int64' (<address>)` at runtime because it is an `Int32`.

That's why the simpler version of your casting code makes me so nervous; it shouldn't actually work, in the same way that the thing you are fixing in this PR (casting `NSNull` to `Foo?`) should never have worked. It seems like sometimes the Swift runtime is doing things that aren't just strictly-interpreted casting operations when casting, on at least some runtimes, but so far I can't find any documentation saying that's actually a thing.

We should only be using casting operations when we're actually following the type hierarchy, and for anything else we should be explicitly *converting* types, not casting them.

Reply:
> the thing here is that all incoming types are Any so they can all be converted directly to int64

I was asking about the dynamic type (i.e. `type(of: args[0])`) rather than static type (`Any`). We can probably find it out by tracing it back to where `args[0]` is set.

Reply:
(I haven't dug into it in depth, but I suspect part of what's going on here has to do with internal details of how various kinds of `NSNumber`s, which are what the Obj-C codec code is actually producing, are being bridged to Swift numeric types, rather than the Swift casting itself.)

Reply:
> but I suspect part of what's going on here has to do with internal details of how various kinds of `NSNumber`s, which are what the Obj-C codec code is actually producing, are being bridged to Swift numeric types

This piqued my curiosity, so I poked around some, and found [this proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0170-nsnumber_bridge.md) which was implemented in Swift 4. `NSNumber` is indeed an odd duck in Swift bridging, such that `as` casting is **not** actually a cast in traditional terms, but is instead doing a conversion so long as the conversion is valid.

So we could rely on this behavior if we check that the type is an `NSNumber` but I would rather we keep the int32/int64 check structure we had before instead since that would continue to work even if someday the engine started using Swift instead (and is also compatible with the NSNumber approach, even if it's more than we technically need).

Reply:
> We can probably find it out by tracing it back to where `args[0]` is set.

The more interesting thing would be to print `type(of:)` to see what the actual underlying runtime type is, which is not necessarily the same what the code at the set point is due to class clusters and bridging. My bet is on the runtime type being `__NSCFNumber` or something closely related to it.

But if you are curious about the code constructing the values, it's going to be [this](https://github.com/flutter/engine/blob/main/shell/platform/darwin/common/framework/Source/FlutterStandardCodecHelper.cc#L118-L129).

Reply:
> The more interesting thing would be to print type(of:) to see what the actual underlying runtime type is

Yeah i think it's worth to confirm what actual type we are dealing with here. From the constructing side, it also looks like NSNumber/__NSCFNumber to me. 

> NSNumber is indeed an odd duck in Swift bridging, such that as casting is not actually a cast in traditional terms, but is instead doing a conversion so long as the conversion is valid.

Interesting finding! 

> I would rather we keep the int32/int64 check structure we had before instead since that would continue to work even if someday the engine started using Swift instead

Nit: we probably need to do `Int32/Int64` pair rather than `Int/Int64` pair in the original code.

Reply:
> Nit: we probably need to do Int32/Int64 pair rather than Int/Int64 pair in the original code.

That change has already been made